better-words/message_parser.py

```diff
@@ -0,0 +1,61 @@
+import yaml
+import re
+from os import listdir
+from os.path import splitext
+
+class MessageParser(object):
+  def __init__(self):
+    self.words = {}
+
+    # iterate over all the word categories and load up each yml file
+    for file in listdir('./better-words/words'):
+      category, extension = splitext(file)
+      if extension == '.yml':
+        with open('./better-words/words/' + file, 'r') as stream:
+          self.words[category] = yaml.load(stream)
+
+  # check if the given message contains a word or phrase match
+  # TODO: handle messages that contain multiple matches
+  # TODO: check if the user has opted out
+  def is_match(self, message):
+    for category in self.words:
+      for entry in self.words[category]['entries']:
+        for word in entry['words']:
+          regex = r'\b' + word + r'\b'
+          match = re.search(regex, message.get('text'))
+          if match:
+            return {
+              'message': message,
+              'category': self.words[category],
+              'entry': entry,
+              'word': match.group(0)
+            }
+    return None
+
+  # format a message to be sent to the user
+  def format(self, match):
+    message = match['message']
+    response = match['category']['response']
+    suggestions = match['entry']['suggestions']
+    word = match['word']
+
+    user = message.get('user')
+    message_text = message.get('text')
+
+    # if the message is multiple lines, reduce the message text
+    # to be quoted to only the line containing the matching word
+    # TODO: replace by linking / sharing the original message
+    # to the user
+    lines = message_text.split('\n')
+    if len(lines) > 1:
+      for l in lines:
+        if word in l: # TODO replace by fancy
```

Comment:
We get the exact word we matched from the regex object here https://github.com/Imperiopolis/better-words/pull/1/files#diff-6c6298e8dfa87bf76960d00c2124f24aR31, but you're right this could still match a partial word.. like if we were searching for `tim`, found it, but then the message also contained `time`. Will add the regex stuff here too.